metagraph/src/annotation/binary_matrix/column_sparse/column_major.cpp

```diff
@@ -109,8 +111,12 @@ bool ColumnMajor::load(std::istream &in) {
 void ColumnMajor::serialize(std::ostream &out) const {
     serialize_number(out, columns_->size());
 
-    for (const auto &column : *columns_) {
+    for (auto &column : *columns_) {
         assert(column.get());
+        // conversion pilfers the converted object, so we place the result back into
+        // #column, to ensure the ColumnMajor instance is valid after serialization
+        const_cast<std::unique_ptr<bit_vector> &>(column)
+                = std::make_unique<bit_vector_sd>(column->convert_to<bit_vector_sd>());
```

Comment:
This should serialize the given vectors and load them from their original representations (we need a loader `static std::unique_ptr<bit_vector> bit_vector::load(filename)`

Not necessary to implement it now, just add a TODO message.

Reply:
It not cool to convert a given representation to `bit_vector_sd`.
It should dump its original representation, which could be bit_vector_stat, bit_vector_rrr, etc.

So the problem here is not in serialization but in loading (we can't load a vector if we don't know its representation).
But this should be solved in the future when we can load a bit vector without necessarily knowing its representation: `bit_vector::load(filename)`. So I asked to add a TODO message to fix this serialization once this loader is implemented.

Why -- to avoid blind transforms to `bit_vector_sd`, which may be inappropriate for certain cases.